benchmark/benchmark_suite.jl

```diff
@@ -60,4 +63,196 @@ function run_benchmark_suite!(suite::BenchmarkSuite)
+
+    write(md, "\n## $(suite.title)\n\n")
+
+    print_NF = any(suite.NF .!= suite.NF[1])
+    print_Grid = any(suite.Grid .!= suite.Grid[1])
+    print_nlat = any(suite.nlat .!= suite.nlat[1]) 
+    print_dynamics = any(suite.dynamics .!= suite.dynamics[1])
+    print_physics = any(suite.physics .!= suite.physics[1])
+
+    column_header = "| Model "
+    column_header *= print_NF ? "| NF " : ""
+    column_header *= "| T "
+    column_header *= "| L "
+    column_header *= print_Grid ? "| Grid " : ""
+    column_header *= print_nlat ? "| Rings " : ""
+    column_header *= print_dynamics ? "| Dynamics " : ""
+    column_header *= print_physics ? "| Physics " : ""
+    column_header *= "| Δt | SYPD | Memory|"
+
+    ncolumns = length(findall('|',column_header)) - 1
+    second_row = repeat("| - ", ncolumns) * "|"
+
+    write(md,"$column_header\n")
+    write(md,"$second_row\n")
+
+    for i in 1:suite.nruns
+
+        row = "| $(suite.model[i]) "
+        row *= print_NF ? "| $(suite.NF[i]) " : ""
+        row *= "| $(suite.trunc[i]) "
+        row *= "| $(suite.nlev[i]) "
+        row *= print_Grid ? "| $(suite.Grid[i]) " : ""
+        row *= print_nlat ? "| $(suite.nlat[i]) " : ""
+        row *= print_dynamics ? "| $(suite.dynamics[i]) " : ""
+        row *= print_physics  ? "| $(suite.physics[i]) " : ""
+
+        Δt = round(Int,suite.Δt[i])
+        sypd = suite.SYPD[i]
+        SYPD = isfinite(sypd) ? round(Int, sypd) : 0
+        memory = prettymemory(suite.memory[i])
+        row *= "| $Δt | $SYPD | $memory |"
+
+        write(md,"$row\n")
+    end
+end 
+
+@kwdef mutable struct BenchmarkSuiteDynamics <: AbstractBenchmarkSuite
+    title::String
+    nruns::Int = 1
+    model::Vector = fill(PrimitiveWetModel, nruns)
+    NF::Vector = fill(SpeedyWeather.DEFAULT_NF, nruns)
+    trunc::Vector{Int} = fill(SpeedyWeather.DEFAULT_TRUNC, nruns)
+    nlev::Vector{Int} = default_nlev(model)
+    Grid::Vector = fill(SpeedyWeather.DEFAULT_GRID, nruns)
+    nlat::Vector{Int} = fill(0, nruns)
+    dynamics::Vector{Bool} = fill(true, nruns)
+    physics::Vector{Bool} = fill(true, nruns)
+    function_names::Vector{String} = default_function_names()
+    time_median::Vector{Vector{Float64}} = [fill(0.0, length(function_names)) for i=1:nruns]
+    memory::Vector{Vector{Int}} = [fill(0, length(function_names)) for i=1:nruns]
+    allocs::Vector{Vector{Int}} = [fill(0, length(function_names)) for i=1:nruns]
+end 
+
+default_function_names() = ["pressure_gradient_flux!", "linear_virtual_temperature!", "temperature_anomaly!", "geopotential!", "vertical_integration!", "surface_pressure_tendency!", "vertical_velocity!", "linear_pressure_gradient!", "vertical_advection!","vordiv_tendencies!", "temperature_tendency!", "humidity_tendency!", "bernoulli_potential!"]
+
+function add_results!(suite::BenchmarkSuiteDynamics, trial::BenchmarkTools.Trial, i_run::Integer, i_func::Integer)
+
+    t = median(trial)
+
+    suite.time_median[i_run][i_func] = t.time 
```

Comment:
Is median what's typically printed with `@btime`? I thought that was lowest?

Reply:
Yes, you are right `@btime` prints the minimum. I didn't know that before. `@benchmark` prints range, median and mean. Any way, not that it matters a lot, but I'd still prefer median.

Reply:
I believe in benchmarking the minimum is often used because it is more robust because that's the most uninterrupted execution of this code